### DISCUSSION TITLE
I'm misunderstanding something about initial configuration of mqtt and batmon

### DISCUSSION THREAD

**Discussion by FFJacot** [fl4p/batmon-ha#162] [Q&A]
Hi,

I'm a home assistant newbie, and probably missing something obvious.

I've installed MQTT, Python Scripts, and batmon and followed the batmon installation write up, but nothing is happening, despite my Raspberry Pi sitting less than a meter away from my 4 batteries. It's not clear if I'm supposed to start anything else, or even where to enter battery configurations: I don't even know if bluetooth on the Pi is active? The Python Scripts is tested and working according to their wtiteup. I haven't done anything to MQTT other than start it.

What must MQTT configuration be?

Lots of confusion and questions here - if you assume I don't know much, you'd be right! Any help in getting going would be appreciated.

**Comment by fl4p** [comment]
did you see https://github.com/fl4p/batmon-ha/blob/master/doc/Standalone.md ?
you basically put your devices and configuration in `options.json` and then run `main.py`

**Reply by FFJacot** [reply]
Thanks for response....

I'm running my Pi with the standard Home assistant configuration, which - to my limited eye - means the only "barebones" access to the Pi I have is via the "Terminal" home assistant add-on, which seems to have set up it's own environment which gives a sort of shell access to do things, I have no idea if the stand alone installation could be done in this environment?

Is there another way of using ssh to get into a Home assistanted Pi? Otherwise I don't see a way of doing the standalone installation.

Note that in the past I have installed Home assistant under docker, which would easily allow standalone, but  docker was a disaster for me as most of the Home assistant stuff I wanted to do was where it was expected to be.

Thanks...

**Reply by FFJacot** [reply]
Unless you're suggesting I run batmon on a separate Pi independent of my Home assistant? I'd like to avoid having too many Pi's in my garage near my inverter set up. I still need to connect up my Axpert type inverter via USB, and supposed this could connect to my Home assistant Pi.....

**Reply by fl4p** [reply]
with HA OS you are pretty much limited to using batmon as an add-on inside a docker container managed by tha HA supervisor.

here's instructions on how to ssh to the vate metal, however HA OS does provide very limited tools https://developers.home-assistant.io/docs/operating-system/debugging/

**Comment by fl4p** [comment]
I agree that using HA addons for the first time can be confusing. just follow the instructions from the readme. you need yo install the mqtt broker addon and the integration. not sure why this isn't already there in Home assistant by default. you can follow any guide you can find, there are no special steps with batmon

**Reply by fl4p** [reply]
If you're keen with linux console, I'd recommend you to ditch the HA OS and start with a fresh Raspberry Pi OS + Home Assistant supervised. Then you can run batmon outside any container. 
If this doesn't sound familiar to you, you’re probably better off with HA OS.

**Reply by FFJacot** [reply]
OK, I'm trying to avoid docker because of the hassles I had earlier trying to install stuff and being unable to find stuff.

I realise I'm trying to do fancyish stuff with very limited HA experience... it'll get easier, but what I'm going through now is flattening the learning curve for later.

I'm OK but a bit rusty with linux console, so maybe what I could do is use my spare Pi, install PI o/s and leave out HA and docker and just install batmon (plus any other future standalone stuff that looks useful). I take it this would work, and the batmon would transfer stuff to MQTT listening on the other HA Pi? If so, then I'll just set up according to the guide for standalone install.

I have MQTT installed and listening on HA Pi.

**Reply by FFJacot** [reply]
The install doc says this should work, so here goes - I'll let you know how things go.

Thanks for the feedback so far - I'm sure there'll be more questions in a day or two.

**Reply by fl4p** [reply]
Yep docker can be annoying especially when dealing with hardware.

**Comment by FFJacot** [comment]
First stumble!

============================================
ffjg@rpib:~` $ cd batmon-ha
ffjg@rpib:~/batmon-ha $ ls
addon_main.sh  CHANGELOG.md  Dockerfile        mqtt_util.py      TODO.md
apparmor.txt   config.yaml   install_bleak.py  README.md         tools
bmslib         doc           main.py           requirements.txt
ffjg@rpib:~/batmon-ha $ python3 --version
Python 3.11.2
ffjg@rpib:~/batmon-ha $ pip3 install -r requirements.txt
error: externally-managed-environment
=============================================

**Reply by FFJacot** [reply]
I'm stuck - after the

pip3 install -r requirements.txt

of the standalone installation procedure, I get 

error: externally-managed-environment

I don't know enough to move further. Any advice?

Thanks

**Reply by FFJacot** [reply]
OK, so

pip3 install -r requirements.txt --break-system-packages

gets a step further. Let's see what the next issue is....

**Reply by FFJacot** [reply]
After changing the location of options.json to /data/options.json and firing up 

python3 main.py

it seems like things are functioning - MQTT on my HA Pi detected my 4 Revov batteries and can display umpteen sensors for each one. As easy as that!?

Now I need to learn about how to set up a flashy dashboard. Are there any examples out there?
